src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -37,6 +37,19 @@ namespace System.Management.Automation
     /// </summary>
     internal abstract class Adapter
     {
+        /// <summary>
+        /// The member type that a callsite binder operates on.
+        /// </summary>
+        /// <remarks>
+        /// An adapter can decide whether the binder can optimize the operations
+        /// on an object member based on the type of the member.
+        /// </remarks>
+        internal enum SiteBinderOperatesOn
```

Comment:
You might consider using the existing flags enum  `System.Management.Automation.PSMemberTypes` instead.

Reply:
I think `System.Reflection.MemberTypes` is a better option :) I updated the code to use that enum type.